system/Files/FileSizeUnit.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Files;
+
+use InvalidArgumentException;
+
+enum FileSizeUnit: int
+{
+    /**
+     * Allows the creation of a FileSizeUnit from Strings like "kb" or "mb"
+     *
+     * @throws InvalidArgumentException
+     */
+    public static function fromString(string $unit): self
+    {
+        return match (strtolower($unit)) {
+            'b'     => self::B,
+            'kb'    => self::KB,
+            'mb'    => self::MB,
+            'gb'    => self::GB,
+            'tb'    => self::TB,
+            default => throw new InvalidArgumentException("Invalid unit: {$unit}"),
+        };
+    }
+
+    case B  = 0;
+    case KB = 1;
+    case MB = 2;
+    case GB = 3;
+    case TB = 4;
```

Comment:
I know this is not restricted by our code style, but it is quite unusual. Can we move it to the top?

Reply:
yes, this should be on top, like constants

Reply:
It cannot be. I had it on the top and either CS Fixer or Rector failed, which is why I moved it to the bottom.

Reply:
Please `composer update` and re-run CS again. This should be fixed by now by the release in coding standard.